core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts

```diff
@@ -106,4 +162,77 @@ export class DotEmaComponent implements OnInit {
             queryParamsHandling: 'merge'
         });
     }
+
+    /**
+     * Handle the dialog close event
+     *
+     * @memberof DotEmaComponent
+     */
+    resetIframeData() {
+        this.store.resetDialog();
+    }
+
+    /**
+     * Create the url to edit a contentlet
+     *
+     * @private
+     * @param {string} inode
+     * @return {*}
+     * @memberof DotEmaComponent
+     */
+    private createEditContentletUrl(inode: string): string {
+        return `/c/portal/layout?p_p_id=content&p_p_action=1&p_p_state=maximized&p_p_mode=view&_content_struts_action=%2Fext%2Fcontentlet%2Fedit_contentlet&_content_cmd=edit&inode=${inode}`;
+    }
+
+    /**
+     * Handle the custom events from the iframe
+     *
+     * @private
+     * @param {Event} event
+     * @memberof DotEmaComponent
+     */
+    private handleNgEvent(event: CustomEvent) {
+        const { detail } = event;
+
+        // Skip the loaded event
+        if (detail.name === NG_CUSTOM_EVENTS.EDIT_CONTENTLET_LOADED) return;
+
+        // This forces a reload in the iframe
+        this.iframe.nativeElement.contentWindow?.postMessage(
+            NOTIFY_CUSTOMER.EMA_RELOAD_PAGE,
+            this.host
+        );
+    }
+
+    /**
+     * Handle the post message event
+     *
+     * @private
+     * @param {{ action: CUSTOMER_ACTIONS; payload: DotCMSContentlet }} data
+     * @return {*}
+     * @memberof DotEmaComponent
+     */
+    private handlePostMessage({
+        origin = this.host,
+        data
+    }: {
+        origin: string;
+        data: { action: CUSTOMER_ACTIONS; payload: DotCMSContentlet };
+    }): () => void {
+        const action = origin !== this.host ? CUSTOMER_ACTIONS.NOOP : data.action;
+
+        return {
+            [CUSTOMER_ACTIONS.EDIT_CONTENTLET]: () => {
+                this.store.patchState({
+                    dialogVisible: true,
+                    dialogHeader: data.payload.title,
+                    dialogIframeLoading: true,
+                    dialogIframeURL: this.createEditContentletUrl(data.payload.inode)
+                });
```

Comment:
Don't call patch state directly, this should be an updater in the store

You don't need to pass the whole thing... just the inode and the title will do and the store can set the url to the iframe, the dialog in true etc...

Reply:
moved to the store